cmd/check_container.go

```diff
@@ -152,111 +152,120 @@ var checkContainerCmd = &cobra.Command{
 			if err != nil {
-				return err
+				return fmt.Errorf(
+					"%w: unable to read file from disk to include in submission: %s: %s",
+					errors.ErrSubmittingToPyxis,
+					rpmManifestFilepath,
+					err,
+				)
 			}
-			defer certImageJsonFile.Close()
 
-			certImageBytes, err := io.ReadAll(certImageJsonFile)
+			var rpmManifest pyxis.RPMManifest
+			err = json.Unmarshal(rpmManifestBytes, &rpmManifest)
 			if err != nil {
-				return err
+				return fmt.Errorf(
+					"%w: data for the %s appears to be malformed: %s",
+					errors.ErrSubmittingToPyxis,
+					"rpm manifest",
+					err,
+				)
 			}
 
-			certImage := new(pyxis.CertImage)
-			err = json.Unmarshal(certImageBytes, &certImage)
+			// Include Preflight's test results in our submission. pyxis.TestResults embeds them.
+			var testResults pyxis.TestResults
+			err = json.Unmarshal(formattedResults, &testResults)
 			if err != nil {
-				return err
+				return fmt.Errorf(
+					"%w: data for the %s appears to be malformed: %s",
+					errors.ErrSubmittingToPyxis,
+					"preflight results.json",
+					err,
+				)
 			}
 
-			certImage.ISVPID = certProject.Container.ISVPID
+			// Populate the certImage with missing values as a result of this execution.
+			certImage.ISVPID = certProject.Container.ISVPID // NOTE: certProject depends on Pyxis calls made outside this block.
 			certImage.Certified = testResults.Passed
 
-			rpmManifestJsonFile, err := os.Open(path.Join(artifacts.Path(), certification.DefaultRPMManifestFilename))
-			if err != nil {
-				return err
-			}
-			defer rpmManifestJsonFile.Close()
-
-			rpmManifestBytes, err := io.ReadAll(rpmManifestJsonFile)
-			if err != nil {
-				return err
-			}
-
-			rpmManifest := new(pyxis.RPMManifest)
-			err = json.Unmarshal(rpmManifestBytes, rpmManifest)
-			if err != nil {
-				return err
-			}
-
+			// Send the preflight logfile as an artifact in our submission.
 			logFileName := viper.GetString("logfile")
-
-			logFile, err := os.Open(logFileName)
+			logFileBytes, logFileSize, err := readFileAndGetSize(logFileName)
 			if err != nil {
-				return err
-			}
-			defer logFile.Close()
-
-			logFileBytes, err := io.ReadAll(logFile)
-			if err != nil {
-				return err
-			}
-
-			logFileInfo, err := logFile.Stat()
-			if err != nil {
-				return err
+				return fmt.Errorf(
+					"%w: unable to read file from disk to include in submission %s: %s",
+					errors.ErrSubmittingToPyxis,
+					logFileName,
+					err,
+				)
 			}
 
 			logFileArtifact := pyxis.Artifact{
 				CertProject: projectId,
 				Content:     base64.StdEncoding.EncodeToString(logFileBytes),
 				ContentType: http.DetectContentType(logFileBytes),
 				Filename:    logFileName,
-				FileSize:    logFileInfo.Size(),
+				FileSize:    logFileSize,
 			}
 
-			artifacts := make([]pyxis.Artifact, 0, 1)
-			artifacts = append(artifacts, logFileArtifact)
+			// additional artifacts to send to Pyxis.
+			artifacts := []pyxis.Artifact{logFileArtifact}
```

Comment:
I'm fine with this for now. We can go back to the prior method when we start adding more artifacts, and we know ahead of time how many there are.